splunk-otel-android/src/main/java/com/splunk/rum/SlowRenderingDetectorImpl.java

```diff
@@ -97,25 +91,20 @@ private static Looper startFrameMetricsLoop() {
 
     @Override
     public void add(Activity activity) {
-        boolean added;
-        synchronized (lock) {
-            added = activities.add(activity);
-        }
-        if (added) {
-            activity.getWindow().addOnFrameMetricsAvailableListener(this, frameMetricsHandler);
+        PerActivityListener listener = new PerActivityListener(activity);
+        PerActivityListener existing = activities.put(activity, listener);
```

Comment:
This is a good call. I actually think this is more important than we both had realized. In my initial `.put()` implementation, the map will be updated with a new listener, but the listener will _not_ be registered. Then later, when we `.remove()` the activity from the map and get the corresponding listener, I suspect the listener will not be removed correctly because of no equals/hashcode. 

In any case, your suggestion is more correct, and even tho it shouldn't really happen, we should do this. 🤘🏻